libs/remote_storage/src/lib.rs

```diff
@@ -313,6 +334,23 @@ impl<Other: RemoteStorage> GenericRemoteStorage<Arc<Other>> {
         }
     }
 
+    // See [`RemoteStorage::list_streaming`].
+    pub fn list_streaming<'a>(
+        &'a self,
+        prefix: Option<&'a RemotePath>,
+        mode: ListingMode,
+        max_keys: Option<NonZeroU32>,
+        cancel: &'a CancellationToken,
+    ) -> impl Stream<Item = Result<Listing, DownloadError>> + 'a {
```

Comment:
given that we already did a Box pin here, we can use a concrete type as the return value (i.e., `Box<dyn Stream>`) instead of doing opaque type inference?

Reply:
I'd like to stay consistent with the `RemoteStorage` trait's interface. Users shouldn't care which specific `Stream` impl is used.